### PR TITLE
Extend `iedb` prefix to multiple semantic subspaces

### PR DESCRIPTION
**Pull request by nagutm**
This PR addresses https://github.com/biopragmatics/bioregistry/issues/1204 by extending the `iedb` prefix into several specific subprefixes to better capture distinct semantic spaces. The subprefixes introduced in this PR are: `iedb.reference`, `iedb.assay`, `iedb.epitope`, `iedb.receptor`, and `iedb.mhc`.

Two main approaches were evaluated for organizing these semantic subspaces:
1. **Retain `iedb` as a parent prefix**: Under this approach, iedb serves as a source of general information, while each subprefix is curated for specific semantic spaces. This structure preserves generic mappings and publication data under the parent iedb prefix, reducing duplication and creating a single reference point for general resources. Each subprefix has a `uri_format` and `pattern` associated with it while the parent prefix does not.

2. **Eliminate the parent prefix and duplicate information**: Alternatively, we could remove the iedb parent prefix entirely and duplicate general information, such as mappings and publications, across each subprefix. While this approach ensures each subprefix is independently comprehensive, it introduces a high amount of redundancy.

This PR implements the first approach, maintaining `iedb` as a parent prefix. The following chages were made:
- General mappings such as `fairsharing`, `integbio`, `re3data` are retained under the parent, while subprefix-specific mappings (`miriam`, `prefixcommons`)  are assigned to its related sematic subprefix.
- Publications that apply broadly across IEDB are preserved under the parent prefix. Any future publications relevant only to specific semantic spaces could be added to the appropriate subprefix as needed.
- `iedb.antigen` is curated separately as a provider for UniProt and ChEBI

One drawback of this approach is that it is not clear what sematic space the parent perfix represents, however as the bioregistry continues to grow larger in the future, I felt that an approach that reduces overall redundancy would be more useful.

### CODE REVIEW

**Comment by nagutm** [comment]
Per https://github.com/biopragmatics/bioregistry/pull/1217#issuecomment-2428413739, I duplicated the publications entry in all subspaces

File: src/bioregistry/data/bioregistry.json [hunk 1]
```diff
@@ -54703,6 +54683,257 @@
...
+      },
+      {
+        "pubmed": "16312048",
+        "title": "The immune epitope database and analysis resource: from vision to blueprint",
+        "year": 2004
+      }
+    ],
+    "uri_format": "https://www.iedb.org/assay/$1"
+  },
+  "iedb.epitope": {
```
Comment (cthoyt) [inline src/bioregistry/data/bioregistry.json:54730]: this record needs to retain the miriam link

Comment (bgyori) [reply src/bioregistry/data/bioregistry.json:54730]: The identifiers.org record appears to be specific to `iedb.reference`, could you elaborate why `iedb.epitope` should retain that link?

Comment (cthoyt) [reply src/bioregistry/data/bioregistry.json:54730]: might be that the diff is too hard to read, it looked like the URI pattern from the MIRIAM record matched the `iedb.epitope` entry. Will take another look later

**Comment by bgyori** [comment]
The test failure is due to the fact that according to strategy 1 outlined by @nagutm above, the parent `iedb` entry is preserved but it doesn't come with an `example` which is required per the test. This all relates to the more general issue at #1222, whether parent entries that don't correspond to a specific semantic space (which would have a specific pattern, uri_format, example, etc.) should be a thing, or in a situation like this with `iedb`, the parent entry should be completely dissolved.

**Comment by nagutm** [comment]
After discussing with @bgyori  we decided that strategy 2 would be better suited for the following reasons:
- Clear semantic spaces: Each subprefix clearly defines a sematic space without any ambiguity as to what the parent prefix represents
- Alignment with current testing standards: This method prevents us from having to modify current established testing standards that require certain fields to be present for a new prefix such as `example`

**Pull request merged**